### PR TITLE
Updated usdt_sample for ubuntu 21.10 (and python3)

### PR DESCRIPTION
**Pull request by bveldhoen**
* Updated readme for Ubuntu 21.10 and python3.
* Readability improvements.
* Using atomic_increment (thus requires bcc 0.21+).
* Using/tested probes created using systemtap (dtrace), next to the
probes created with macros in the folly header.
* Also tested while building usdt_sample with clang.
* Some (modern) cmake changes.
* Added script to execute build and sample python scripts.

### CODE REVIEW

**Comment by yonghong-song** [comment]
@davemarchevsky could you take a look?

File: examples/usdt_sample/scripts/bpf_text_shared.c [hunk 1]
```diff
@@ -4,15 +4,28 @@
...
-    for (int i = 0; i < sizeof(needle) - 1; ++i) {
-        if (needle[i] != haystack[i]) {
+    static const char* null_ptr = 0x0;
+    static const char null_terminator = '\0';
+
+    static const char filter_string[] = "FILTER_STRING"; ///< The filter string is replaced by python code.
+    if (null_ptr == inputString) {
+        return false;
+    }
+    // bpf_trace_printk("inputString: '%s'", inputString);
```
Comment (davemarchevsky) [inline examples/usdt_sample/scripts/bpf_text_shared.c:18]: could you remove the commented-out printk?

File: examples/usdt_sample/scripts/lat_avg.py [hunk 2]
```diff
@@ -12,23 +12,25 @@
     formatter_class=argparse.RawDescriptionHelpFormatter)
 parser.add_argument("-p", "--pid", type=int, help="The id of the process to trace.")
 parser.add_argument("-i", "--interval", type=int, help="The interval in seconds on which to report the latency distribution.")
-parser.add_argument("-c", "--count", type=int, default=16, help="The count of samples over which to calculate the moving average.")
+parser.add_argument("-m", "--maxsamplesize", type=int, default=16, help="The maximum number of samples over which to calculate the moving average.")
```
Comment (davemarchevsky) [inline examples/usdt_sample/scripts/lat_avg.py:15]: could we keep `-c`, `--count` here, and add `-m`, `--maxsamplesize` as additional aliases?

**Review by davemarchevsky — COMMENTED** [review]
aside from nits LGTM

**Review by davemarchevsky — CHANGES_REQUESTED** [review]
see requested nit changes above, accidentally marked previous review as a 'comment'

File: examples/usdt_sample/scripts/bpf_text_shared.c (hunk 1 above)
Comment (bveldhoen) [reply examples/usdt_sample/scripts/bpf_text_shared.c]: Done

File: examples/usdt_sample/scripts/lat_avg.py (hunk 2 above)
Comment (bveldhoen) [reply examples/usdt_sample/scripts/lat_avg.py]: Done, put --count back, removed --maxsamplesize.

**Comment by bveldhoen** [comment]
Updated the underlying commit, please re-review.

**Pull request merged**